supplementary/alica_ros1/alica_ros_turtlesim/libalica-turtlesim/src/GoToCalculatedResult.cpp

```diff
@@ -0,0 +1,46 @@
+#include "GoToCalculatedResult.h"
+
+#include <constraintsolver/CGSolver.h>
+#include <engine/logging/Logging.h>
+
+#include <memory>
+
+namespace turtlesim
+{
+
+GoToCalculatedResult::GoToCalculatedResult(alica::BehaviourContext& context)
+        : BasicBehaviour(context)
+{
+}
+
+void GoToCalculatedResult::run()
+{
+    // solve constraints and get value
+    if (!_query.getSolution<alica::reasoner::CGSolver, double>(getPlanContext(), _results)) {
+        alica::Logging::logWarn(alica::LOGNAME) << "Behaviour: " << getName() << " calculating ...";
+        return;
+    }
+
+    if (_turtle->moveTowardPosition(_results[0], _results[1])) {
+        setSuccess(); // set success if turtle reach goal
+    }
+}
```

Comment:
I think GoToCalculatedResult should be refactored into 2 states. 

One is the calculation of the goal and the other is the actual GoTo behavior. These ops are already synchronous here in this function (first we calculate the target and just later we move). It would be easier to debug and feels more natural to have one state just calculating the target and the other doing the actual movement. 

We already have GoTo behavior, we could just reuse it for the other state.

Reply:
I thought of this but actually the result can be recalculated many times, for example if a new turtle joins the formation.

Reply:
The same could be achieved with these states (considering we have split) by having a WaitForTrigger behavior on both states triggerring when a new turtle joins, and a CustomAnyChildSuccess to a failure state coming from both states, to simulate an ActionServer and make MakeFormation retry instantly for this turtle when this happens.

I say this because from a robotics point of view, it would be better for me if I had traces specific for each state so I know where some bug comes from, and the more these states are modularized and express a single responsibility node, the better since the states will be more meaningful to whom is working with it. 

I understand this is turtlesim and is just a tutorial thing, but since we are changing this, it would be a good case for teaching someone. If you think it is too much we can do in another PR, I just fear it gets lost in the backlog.

Reply:
Anyway, I fear we add logic to just one state when we can split them when the goal is to use the full capability of having a state machine controlling the behavior of our robots. Maybe I'm overreacting but it's just a thought.

Reply:
>The same could be achieved with these states (considering we have split) by having a WaitForTrigger behavior on both states triggering when a new turtle joins, and a CustomAnyChildSuccess to a failure state coming from both states, to simulate an ActionServer and make MakeFormation retry instantly for this turtle when this happens (I don't know how to do this last one tbh).

So currently the only way turtles know about each other is through the alica solver - they cannot listen to WaitForTrigger topics of other turtles (and doing so would also not be reliable).  It is possible to reuse the GoTo state, but I think it would add complexity overall.  GoTo is a simple navigation behavior that just goes to one point and then marks success.  This makes it useful for simpler navigation flows, but does not work as well for situations where the goal keeps changing and reaching the goal once does not guarantee that you don't need to navigate again.

For such situations, we can make a new reusable navigation plan that handles changing goals and uses GoTo internally, but I think that is out of scope here.